cmds/rm/rm.go

```diff
@@ -11,81 +11,166 @@
 //     -i: interactive mode
 //     -v: verbose mode
 //     -R: remove file hierarchies
-//     -r: equivalent to -R
+
 package main
 
 import (
 	"bufio"
+	"errors"
 	"flag"
 	"fmt"
+	"io/ioutil"
 	"os"
-	"path"
+	"path/filepath"
 )
 
-var (
-	flags struct {
-		r bool
-		v bool
-		i bool
+type rmFlags struct {
+	recursive   bool
+	verbose     bool
+	interactive bool
+}
+
+func isEmpty(file string) (bool, error, []os.FileInfo) {
+	fileList, err := ioutil.ReadDir(file)
+	if err != nil {
+		//ignore this boolean return value
+		return false, err, fileList
 	}
-	cmd = "rm [-Rrvi] file..."
-)
+	if len(fileList) == 0 {
+		return true, nil, fileList
```

Comment:
I'm not sure what the bool means. I'm not sure this function is very useful -- see comment on usage below.